test/e2e/aks_tags.go

```diff
@@ -136,14 +131,16 @@ func AKSAdditionalTagsSpec(ctx context.Context, inputGetter func() AKSAdditional
 		}, inputGetter().WaitForUpdate...).Should(Succeed())
 		Eventually(checkTags, input.WaitForUpdate...).Should(Succeed())
 
-		By("Restoring initial tags for control plane")
-		expectedTags = initialTags
-		Eventually(func(g Gomega) {
-			g.Expect(mgmtClient.Get(ctx, client.ObjectKeyFromObject(infraControlPlane), infraControlPlane)).To(Succeed())
-			infraControlPlane.Spec.AdditionalTags = expectedTags
-			g.Expect(mgmtClient.Update(ctx, infraControlPlane)).To(Succeed())
-		}, inputGetter().WaitForUpdate...).Should(Succeed())
-		Eventually(checkTags, input.WaitForUpdate...).Should(Succeed())
+		if initialTags != nil {
```

Comment:
ASO does not support deleting all tags: https://github.com/Azure/azure-service-operator/issues/3367